controller/Previewer.php

```diff
@@ -193,23 +215,55 @@ public function getItem()
                     ->loadCompiledItemData();
 
                 $response['baseUrl'] = $itemPreviewer->getBaseUrl();
-
-            } else if ($itemUri) {
-                // Load RESOURCE item data
-                // TODO
+            } elseif ($itemUri) {
+                $item = $this->getResource($itemUri);
+                $lang = $this->getSession()->getDataLanguage();
+                $packer = new Packer($item, $lang);
+                $packer->setServiceLocator($this->getServiceLocator());
+
+                /** @var ItemPack $itemPack */
+                $itemPack = $packer->pack();
+                $response['content'] = $itemPack->JsonSerialize();
+                $response['baseUrl'] = _url('asset', null, null, ['uri' => $itemUri, 'path' => '/']);
             } else {
-                throw new \common_exception_BadRequest('Either itemUri or resultId needs to be provided.');
+                throw new BadRequestException('Either itemUri or resultId needs to be provided.');
             }
 
             $response['success'] = true;
-        } catch (\Exception $e) {
+        } catch (Exception $e) {
             $response = $this->getErrorResponse($e);
             $code = $this->getErrorCode($e);
         }
 
         $this->returnJson($response, $code);
     }
 
+    /**
+     * Gets access to an asset
+     * @throws \common_exception_Error
+     * @throws FileNotFoundException
+     * @throws CommonException
+     * @throws Exception
+     */
+    public function asset()
+    {
+        $requestParams = $this->getPsrRequest()->getQueryParams();
```

Comment:
please check that the $this->getPsrRequest()->getQueryParams() does no do the decode already 
because i can see the ` $path = rawurldecode($requestParams['path']);`